tests/e2e/mandatory_module_test.go

```diff
@@ -0,0 +1,203 @@
+					Should(Equal(ErrNotFound))
+			})
+			By("And the KCP Kyma CR is in a \"Ready\" State", func() {
+				Consistently(KymaIsInState).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+					Should(Succeed())
+			})
+		})
+
+		It("When the mandatory Manifest skip reconciliation label is removed", func() {
+			Eventually(SetSkipLabelToMandatoryManifests).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient, false).
+				Should(Succeed())
+
+			By("Then mandatory SKR Module Default CR is recreated", func() {
+				Eventually(CheckIfExists).
+					WithContext(ctx).
+					WithArguments("sample-yaml", "kyma-system",
+						"operator.kyma-project.io", "v1alpha1", "Sample", runtimeClient).
+					Should(Succeed())
+			})
+			By("Then mandatory SKR Module Deployment is recreated", func() {
+				Eventually(DeploymentIsReady).
+					WithContext(ctx).
+					WithArguments(runtimeClient, deployName,
+						TestModuleResourceNamespace).
+					Should(Succeed())
+			})
+
+			By("And the KCP Kyma CR is in a \"Ready\" State", func() {
+				Consistently(KymaIsInState).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+					Should(Succeed())
+			})
+		})
+
+		It("When mandatory Module is enabled on SKR Kyma CR", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, v1beta2.Module{
+					Name:    "template-operator",
+					Channel: "regular",
+				}).
+				Should(Succeed())
+			By("Then Kyma is in a \"Error\" State", func() {
+				Eventually(KymaIsInState).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateError).
+					Should(Succeed())
+			})
+			By("When mandatory Module is disabled on SKR Kyma CR", func() {
+				Eventually(DisableModule).
+					WithContext(ctx).
+					WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, "template-operator").
+					Should(Succeed())
+			})
+			By("Then Kyma is back in a \"Ready\" State", func() {
+				Eventually(KymaIsInState).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+					Should(Succeed())
+			})
+		})
+
+		It("When new version of ModuleTemplate is applied", func() {
+			cmd := exec.Command("kubectl", "apply", "-f",
+				"../moduletemplates/mandatory_moduletemplate_template_operator_v2.yaml")
+			out, _ := cmd.CombinedOutput()
+			//			Expect(err).NotTo(HaveOccurred())
+			GinkgoWriter.Printf(string(out))
+			By("Then Kyma mandatory Module is updated on SKR Cluster", func() {
+				Eventually(DeploymentIsReady).
+					WithContext(ctx).
+					WithArguments(runtimeClient, "template-operator-v2-controller-manager",
+						TestModuleResourceNamespace).
+					Should(Succeed())
+			})
+			By("And old Module Operator Deployment is removed", func() {
```

Comment:
At the moment it's just testing the old things are removed. How about testing that new things are created as expected?

Reply:
It is testing that new deployment is created, looks at the By statements above 

`Then Kyma mandatory Module is updated on SKR Cluster`